### DISCUSSION TITLE
Adding new empty row and editing all cells one by one

### DISCUSSION THREAD

**Discussion by kenyazilim** [vitmalina/w2ui#2364] [Q&A]
I am adding a new row to the grid and then using grid.editField(newRecId, 1) to edit the cell after the recid (recid is hidden) so that the user can begin to enter data into the new row. I want to be able to allow the user to enter data into the first cell and then, when they press the Enter key, move on to edit the next cell and so on until the last cell. The editField function does not seem to work after the Enter key is pressed. If the new row is added at the top, it moves to the row below, but if it is added at the bottom, it remains in place. I have managed to detect that the key pressed is 'Enter', but I can't seem to move to the second cell. How can I achieve this?

**I'm using version 2 of w2ui**

```
let grid = new w2grid({
	.....
	.....	
	, onChange: function (event) {
		// et the Enter key
		let basilanTus = event.detail.originalEvent.key;
		
		event.onComplete = function () {
			.....
			.....

			//this is working
			if (basilanTus === 'Enter') {
				grid.editField(recid, 2);
			}
			.....
			.....
		}
	}	
	.....
	.....

});

grid.hideColumn('recid');

function addNewEmptyRow() {
	yeniKayit = true;

	const now = new Date();
	const year = now.getFullYear();
	const month = now.getMonth();
	const day = now.getDate();
	const hours = now.getHours();
	const minutes = now.getMinutes();
	const seconds = now.getSeconds();

	let yeniRecId = year + '' + month + '' + day + '' + hours + '' + minutes + '' + seconds;

	let yeniRowJson = {
		recid: yeniRecId
	};

	var columns = grid.columns;
	
	for (var i = 1; i < columns.length; i++) {

		yeniRowJson[columns[i].field] = ''
	}
	
	grid.add(yeniRowJson);
	
	//this is working
	grid.editField(yeniRecId, 1);
}
```

**Comment by vitmalina** [comment]
You need to add another empty record to the grid if it was the last record that was changed.
